sda-commons-server-kafka/src/main/java/org/sdase/commons/server/kafka/consumer/strategies/retryprocessingerror/RetryCounter.java

```diff
@@ -0,0 +1,65 @@
+package org.sdase.commons.server.kafka.consumer.strategies.retryprocessingerror;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+
+/**
+ * Helper class counting the number of executions for a given record.
+ *
+ * <p>Internally it increases a counter for the current offset of the record as long as the offset
+ * remains the same. If the offset changes, the counter is reset to 0.
+ */
+public class RetryCounter {
```

Comment:
Thanks. For me it's easier to understand although the implementation is basically the same.